app/scripts/lib/ppom/ppom-middleware.ts

```diff
@@ -78,9 +81,13 @@ export function createPPOMMiddleware<
       const securityAlertsEnabled =
         preferencesController.store.getState()?.securityAlertsEnabled;
 
-      const { chainId } = getProviderConfig({
-        metamask: networkController.state,
-      });
+      // This will always exist as the SelectedNetworkMiddleware
+      // adds networkClientId to the request before this middleware runs
+      const { chainId } =
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        networkController.getNetworkConfigurationByNetworkClientId(
+          req.networkClientId,
+        )!;
```

Comment:
what is the bang for here again?

Reply:
Also do we need some error handling in case this request fails somehow?

Reply:
bang tells typescript that this result will always be defined (vs `undefined`). If the SelectedNetworkMiddleware puts an invalid networkClientId on the request, I think it's probably fine if this blows up trying to dereference chainId since we have bigger issues. 

If we were to do error handling, it seems like we should throw earlier in the middleware stack? Perhaps in the SelectedNetworkMiddleware itself?